api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityAndEntitlementService.java

```diff
@@ -97,17 +97,30 @@ private EligibilityAndEntitlementDecision buildDecision(EligibilityResponse elig
                                                             PaymentCycleVoucherEntitlement entitlement,
                                                             UUID existingClaimId) {
         EligibilityStatus eligibilityStatus = determineEligibilityStatus(eligibilityResponse, entitlement);
+        PaymentCycleVoucherEntitlement voucherEntitlement = determinePaymentCycleVoucherEntitlementFromStatus(entitlement, eligibilityStatus);
```

Comment:
Could we move this to before `EligibilityStatus eligibilityStatus = determineEligibilityStatus(eligibilityResponse, entitlement);`?
That way we still include the voucher entitlement if the reason we decided they were ineligible is because that entitlement showed zero vouchers (i.e. the eligibility service said they were eligible, but we calculated zero vouchers)

Reply:
Why would we want to store the voucher entitlement if they are deemed to be INELIGIBLE because the entitlement has no vouchers in it? If we were going to do that then we might as well store an empty voucher entitlement for all scenarios otherwise we'll be inconsistent.

Reply:
My thinking was: to show our working out. If we're not going to do that, then  I think we need to make sure we're recording that they weren't entitled to any vouchers, sowe need to set totalVouchers and totalEntitlementAmountInPence to zero (we can do that in `PaymentCycle.applyVoucherEntitlement` if the voucher is null).

Reply:
As discussed, probably the best way to handle this is to add an `IneligibilityReason` enum & field. So this PR goes in as is, and we raise a story to store the reason for ineligibility.

Reply:
Yep, that's a good plan.